src/base/utils/os.cpp

```diff
@@ -0,0 +1,390 @@
+
+        return keys;
+    }
+
+    QString getRegValue(HKEY handle, const QString &name = QString())
+    {
+        QString result;
+
+        DWORD type = 0;
+        DWORD cbData = 0;
+        LPWSTR lpValueName = NULL;
+        if (!name.isEmpty()) {
+            lpValueName = new WCHAR[name.size() + 1];
+            name.toWCharArray(lpValueName);
+            lpValueName[name.size()] = 0;
+        }
+
+        // Discover the size of the value
+        ::RegQueryValueExW(handle, lpValueName, NULL, &type, NULL, &cbData);
+        DWORD cBuffer = (cbData / sizeof(WCHAR)) + 1;
+        LPWSTR lpData = new WCHAR[cBuffer];
+        LONG res = ::RegQueryValueExW(handle, lpValueName, NULL, &type, (LPBYTE)lpData, &cbData);
+        if (lpValueName)
+            delete[] lpValueName;
+
+        if (res == ERROR_SUCCESS) {
+            lpData[cBuffer - 1] = 0;
+            result = QString::fromWCharArray(lpData);
+        }
+        delete[] lpData;
+
+        return result;
+    }
+
+    QString pythonSearchReg(const REG_SEARCH_TYPE type)
+    {
+        HKEY hkRoot;
+        if (type == USER)
+            hkRoot = HKEY_CURRENT_USER;
+        else
+            hkRoot = HKEY_LOCAL_MACHINE;
+
+        REGSAM samDesired = KEY_READ;
+        if (type == SYSTEM_32BIT)
+            samDesired |= KEY_WOW64_32KEY;
+        else if (type == SYSTEM_64BIT)
+            samDesired |= KEY_WOW64_64KEY;
+
+        QString path;
+        LONG res = 0;
+        HKEY hkPythonCore;
+        res = ::RegOpenKeyExW(hkRoot, L"SOFTWARE\\Python\\PythonCore", 0, samDesired, &hkPythonCore);
+
+        if (res == ERROR_SUCCESS) {
+            QStringList versions = getRegSubkeys(hkPythonCore);
+            qDebug("Python versions nb: %d", versions.size());
+            versions.sort();
+
+            bool found = false;
+            while (!found && !versions.empty()) {
+                const QString version = versions.takeLast() + "\\InstallPath";
+                LPWSTR lpSubkey = new WCHAR[version.size() + 1];
+                version.toWCharArray(lpSubkey);
+                lpSubkey[version.size()] = 0;
+
+                HKEY hkInstallPath;
+                res = ::RegOpenKeyExW(hkPythonCore, lpSubkey, 0, samDesired, &hkInstallPath);
+                delete[] lpSubkey;
+
+                if (res == ERROR_SUCCESS) {
+                    qDebug("Detected possible Python v%s location", qUtf8Printable(version));
+                    path = getRegValue(hkInstallPath);
+                    ::RegCloseKey(hkInstallPath);
+
+                    if (!path.isEmpty() && QDir(path).exists("python.exe")) {
+                        qDebug("Found python.exe at %s", qUtf8Printable(path));
+                        found = true;
+                    }
+                }
+            }
+
+            if (!found)
+                path = QString();
+
+            ::RegCloseKey(hkPythonCore);
+        }
+
+        return path;
+    }
+
+    QString getCurrentAssoc(const QString &cls)
```

Comment:
Is there any other function except "Current"?

Reply:
Ok, will change it.